src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -1990,6 +1990,16 @@ static void recv_trailing_metadata_ready(void* arg, grpc_error* error) {
   }
   // Not retrying, so commit the call.
   retry_commit(elem, retry_state);
+  // Now that the try is committed, give the trailer to the lb policy as needed
+  if (calld->pick.recv_trailing_metadata_ready != nullptr) {
```

Comment:
This code block handles the case where retries are enabled, but we also need to handle the case where retries are not enabled.  In that case, we need to write a separate callback to intercept recv_trailing_metadata_ready that duplicates this code block.

This should probably be done similarly to how we were handling channelz prior to #16827.